dask/async.py

```diff
@@ -488,12 +488,10 @@ def fire_task():
                 task = dsk[key]
                 _execute_task(task, data)  # Re-execute locally
             else:
-                raise type(res)('\nRemote Exception:\n'
-                              + '-----------------\n'
-                              + str(res) + '\n\n'
-                              + 'Traceback:\n'
-                              + '----------\n'
-                              + tb)
+                print("Traceback:\n"
+                      "----------")
+                print(tb)
```

Comment:
For what you suggested, a caching version of something like this should work:

```python
def make_remote_exception(cls):
    def __init__(self, args, traceback):
        cls.__init__(self, *args)
        self.traceback = traceback
    def __str__(self):
        return ('\n\nOriginal Error:\n'
                '--------------\n'
                '{name}: {err}\n\n'
                'Traceback:\n'
                '----------\n'
                '{tb}\n').format(name=cls.__name__, err=cls.__str__(self),
                                 tb=self.traceback)
    return type('Remote' + cls.__name__, (cls,), {'__init__': __init__, '__str__': __str__})
```

In use:

```python
In [136]: r = make_remote_exception(type(e))

In [137]: e
Out[137]: UnicodeEncodeError('charmap', u'\u0411', 0, 1, 'character maps to <undefined>')

In [138]: t = r(e.args, 'This is a\ntraceback')

In [139]: raise t
---------------------------------------------------------------------------
RemoteUnicodeEncodeError                  Traceback (most recent call last)
<ipython-input-139-02b421b9d115> in <module>()
----> 1 raise t

RemoteUnicodeEncodeError:

Original Error:
--------------
UnicodeEncodeError: 'charmap' codec can't encode character u'u0411' in position 0: character maps to <undefined>

Traceback:
----------
This is a
traceback


In [140]: t.
t.args       t.encoding   t.end        t.message    t.object     t.reason     t.start      t.traceback
```